### PR TITLE
gltfio: add 'detach' methods to allow ownership transfer

### PR DESCRIPTION
**Pull request by prideout**
These new methods allow gltfio to be integrated into internal Google
libraries.

### CODE REVIEW

**Comment by prideout** [comment]
I'd like to cherry pick this to RC so I added it to section v1.25.5 in the release notes.

File: libs/gltfio/include/gltfio/FilamentAsset.h [hunk 1]
```diff
@@ -334,6 +334,23 @@ class UTILS_PUBLIC FilamentAsset {
...
             SceneMask sceneFilter);
 
+    /**
+     * Releases ownership of entities and their Filament components.
+     *
+     * This makes the client take responsibility for destroying Filament
+     * components (e.g. Renderable, TransformManager component) as well as
+     * the underlying entities.
+     */
+    void detachFilamentComponents();
```
Comment (romainguy) [inline libs/gltfio/include/gltfio/FilamentAsset.h:344]: Is the name "Filament" required here?

**Review by romainguy — APPROVED** [review]

File: libs/gltfio/include/gltfio/FilamentAsset.h (hunk 1 above)
Comment (prideout) [reply libs/gltfio/include/gltfio/FilamentAsset.h:344]: Imp uses our ECS, this makes it clear that Imp components are not being detached.

Comment (romainguy) [reply libs/gltfio/include/gltfio/FilamentAsset.h:344]: 👍

**Pull request merged**